File: maximum-subarray/maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ret = nums[0]
        cur = 0
        for i in nums:
            if cur < 0:
                cur = 0
            cur += i
            ret = max(ret, cur)
        return ret
        

File: maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int ret = nums[0];
        int cur = 0;
        for (int i : nums) {
            if (cur < 0) cur = 0;
            cur += i;
            ret = Math.max(ret, cur);
        }
            
        return ret;
    }
}